### PR TITLE
AuthorshipReporter: add support for incremental authorship build #135

### PR DESCRIPTION
**Pull request by yamidark**
Part of #135

### CODE REVIEW

File: src/main/java/reposense/authorship/FileInfoExtractor.java [hunk 1]
```diff
@@ -92,4 +219,11 @@ public static FileInfo generateFileInfo(String repoRoot, String relativePath) {
     private static boolean shouldIgnore(String name, List<String> ignoreList) {
         return ignoreList.stream().anyMatch(name::contains);
     }
+
+    /**
+     * Returns true if the {@code filePath}'s extension is inside the white list.
+     */
+    private static boolean isFileWhiteListed(String filePath) {
+        return filePath.endsWith(".java") || filePath.endsWith(".adoc");
```
[the remark this answers is not in the document]
Comment (yong24s) [reply src/main/java/reposense/authorship/FileInfoExtractor.java]: #159 has a list of user-supplied formats, for you to whitelist, please.

[the remark this answers is not in the document]
Comment (yamidark) [reply src/main/java/reposense/authorship/FileInfoExtractor.java]: Will be integrated with the other components later to include the other hardcoded default file types and user-supplied formats

**Comment by yamidark** [comment]
@eugenepeh For your review.

**Comment by eugenepeh** [comment]
sorry about this PR, seems like it was automatically closed when I update the incremental branch. 
Please create a new one as I am unable to reopen this.

**Pull request closed without merging**